### ISSUE TITLE
can't configure endpoint for testing or proxying

### ISSUE THREAD

**Issue by jackphel** [twilio/twilio-java#203]
please add another constructor to https://github.com/twilio/twilio-java/blob/master/src/main/java/com/twilio/sdk/TwilioRestClient.java#L18
that takes an endpoint.

e.g.
```
	public TwilioRestClient(final String accountSid, final String authToken, final String endpoint) {
		super(accountSid, authToken, endpoint);

		authAccount = new Account(this);
		authAccount.setSid(accountSid);
		authAccount.setAuthToken(authToken);
	}
```

I run tests against my server container giving rest inputs and then validating rest outputs.

I want to stub twilio as a service and resolving the host "twilio" in dns configuration to api.twilio for prod or a local test service  for dev makes this super easy without changing anything in my application code. I run  docker container with `--add-host twilio:MY_TEST_REQUEST_LOGGER` which points to my stub server that just collects and exposes all the requests made to it so that another container can query my app server and then query this stub server to verify the app server's output.

**Comment by skimbrel** [comment]
This has been available since 3.7.2: https://github.com/twilio/twilio-java/blob/master/src/main/java/com/twilio/sdk/TwilioRestClient.java#L26

**Comment by jackphel** [comment]
XD haha I didn't even look below the file I copied. Thanks for the help.
